### ISSUE TITLE
Update dependencies

### ISSUE THREAD

**Issue by jdussouillez** [quarkiverse/quarkus-jooq#89]
There are some updates of dependencies pending in the MRs (https://github.com/quarkiverse/quarkus-jooq/pull/81, https://github.com/quarkiverse/quarkus-jooq/pull/87, https://github.com/quarkiverse/quarkus-jooq/pull/88) that would be nice to have because they're a bit old.

- This plugin still uses Quarkus v2.5.1 (released in December 2021)
- Also it still uses jOOQ v3.15.5 (released in December 2021). The v3.17.3 contains a lot of small fixes that could be very useful for us.

Thanks

**Comment by jdussouillez** [comment]
Any updates ?

**Comment by escay** [comment]
A dependency problem when using jOOQ with Quarkus 3.0.0.Alpha2 I run into is the javax to jakarta package change.
My jOOQ code compiles, but running the Quarkus code fails on runtime with:

```
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkiverse.jooq.deployment.JooqProcessor#build threw an exception: java.lang.NoClassDefFoundError: javax/enterprise/context/ApplicationScoped
	at io.quarkiverse.jooq.deployment.JooqProcessor.createDslContextProducerBean(JooqProcessor.java:117)
	at io.quarkiverse.jooq.deployment.JooqProcessor.build(JooqProcessor.java:83)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
	at java.base/java.lang.Thread.run(Thread.java:1589)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```
This might relate to issue #30 where it is stated that the latest jOOQ version no longer depends on javax packages.
(In the mean time Quarkus 3 is moving to JPA 3.1: https://github.com/quarkusio/quarkus/issues/26986)

So perhaps updating the code to jOOQ v3.16 or newer and changing this plugin code to no longer use javax in JooqProcessor might solve this item as well.

**Comment by jdussouillez** [comment]
Released in [v0.4.0](https://github.com/quarkiverse/quarkus-jooq/releases/tag/0.4.0)
